docs/intents.md

```diff
@@ -0,0 +1,404 @@
+[Table of contents](README.md#table-of-contents)
+
+# Intents
+
+- [Overview](#overview)
+- [Glossary](#glossary)
+- [Proposal](#proposal)
+    - [1. Manifest](#1-manifest)
+    - [2. Intent Start](##2-intent-start)
+    - [3. Service Resolution](#3-service-resolution)
+    - [4. Handshake](#4-handshake)
+    - [5. Processing & Terminating](#5-processing--terminating)
+- [Annex](#annex)
+    - [Use cases](#use-cases)
+    - [Bibliography & Prior Art](#bibliography--prior-art)
+    - [Discarded Ideas](#discarded-ideas)
+
+
+## Overview
+
+A typical Cozy Cloud runs multiple applications, but most of these applications are focused on one task and interact with one particular type of data.
+
+However, Cozy Cloud especially shines when data is combined across apps to provide an integrated experience. This is made difficult by the fact that apps have no dedicated back-end and that they have restricted access to documents.
+
+This document outlines a proposal for apps to rely on each other to accomplish certain tasks and gain access to new documents, in a way that hopefully is neither painful for the users or the developers.
```

Comment:
s/neither painful for the users or the developers/neither painful for the users nor the developers/